### PR TITLE
4.x: Update to static content

### LINKED ISSUE #9471: 4.x: Static content from classpath improvements
Currently we generate temporary files for extracted jar entries as `ws-****.je`

A few improvements could be done:
- use `helidon-` (or `helidon-ws`) prefix to clearly bind them to Helidon
- add configuration option to cache in-memory only (probably to a certain size), to avoid creating the files altogether

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #9471 

### Description

This PR introduces static content Server feature. This allows us to configure static content in config, with multiple handlers supported (both for `path` and `classpath` based static content).
The new configuration is using Helidon builders, and I have added the Config metadata plugin to have the reference available.
Old approach is deprecated and marked for removal, usages are changed to new approach, including MP.

### Documentation

Documentation updated both for SE and MP

### CODE REVIEW

File: docs/src/main/asciidoc/mp/server.adoc [hunk 1]
```diff
@@ -333,24 +333,41 @@ io.helidon.examples.AdminService:
...
+server.features.static-content.classpath.0.location=/WEB
 # default is index.html
-server.static.classpath.welcome=resource.html
+server.features.static-content.classpath.0.welcome=resource.html
 # static content path - default is "/"
-# server.static.classpath.context=/static-cp
+# server.features.static-content.classpath.0.context=/static-cp
+----
+
+It is usually easier to configure list based options using `application.yaml` instead, such as:
```
Comment (spericas) [inline docs/src/main/asciidoc/mp/server.adoc:354]: list-based?

File: docs/src/main/asciidoc/se/webserver.adoc [hunk 2]
```diff
@@ -626,12 +626,12 @@ To enable HTTP/2 support add the following dependency to your project's `pom.xml
...
 
-+Use the `io.helidon.webserver.staticcontent.StaticContentService` class to serve files and classpath resources.
-`StaticContentService` can be created for any readable directory or classpath
-context root and registered on a path in `HttpRouting`.
+Static content is served through a `StaticContentFeature`. As with other server features, it can be configured through config,
+or registered with server config builder.
 
-You can combine dynamic handlers with `StaticContentService` objects: if no file matches the request path, then the request is forwarded to
-the next handler.
+Static content supports serving of files from classpath, or from any readable directory on file system.
```
Comment (spericas) [inline docs/src/main/asciidoc/se/webserver.adoc:632]: on the

File: docs/src/main/asciidoc/se/webserver.adoc [hunk 3]
```diff
@@ -626,12 +626,12 @@ To enable HTTP/2 support add the following dependency to your project's `pom.xml
...
+Each content handler has a location specified (required), and can provide a context that will be registered with the WebServer
```
Comment (spericas) [inline docs/src/main/asciidoc/se/webserver.adoc:633]: handler must include a location, and can optional provide ...

File: webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/BaseHandlerConfigBlueprint.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    /**
+     * Configure capacity of cache used for resources. This cache will make sure the media type and location is discovered
+     * faster.
+     * <p>
+     * To cache content (bytes) in memory, use {@link io.helidon.webserver.staticcontent.BaseHandlerConfig#memoryCache()}
+     *
+     * @return maximal number of cached records, only caches media type and Path, not the content
+     */
+    @Option.Configured
+    Optional<Integer> recordCacheCapacity();
```
Comment (spericas) [inline webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/BaseHandlerConfigBlueprint.java:136]: Shouldn't we use `max` as a prefix here?

File: webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Maximal capacity of the cached bytes of file content.
+     * If set to {@code 0}, the cache is unlimited. To disable caching, set {@link #enabled()} to {@code false},
+     * or do not configure a memory cache at all.
+     *
+     * @return maximal number of bytes in cache
+     */
+    @Option.DefaultLong(50_000_000)
+    @Option.Configured
+    long maxBytes();
```
Comment (spericas) [inline webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java:48]: Maybe use KBs here?

File: docs/src/main/asciidoc/mp/server.adoc (hunk 1 above)
Comment (tomas-langer) [reply docs/src/main/asciidoc/mp/server.adoc:354]: such as:
```yaml
server.features.static-content.classpath:
 - context:  "/static"
   location: "/web"
 - context: "/js"
   location: "/js"
```
i.e. a list of classpath configuration for static content. In properties you must use index, such as:
```properties
server.features.static-content.classpath.0.context=/static
server.features.static-content.classpath.0.location=/web
server.features.static-content.classpath.1.context=/js
server.features.static-content.classpath.1.location=/js
```

File: webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java (hunk 5 above)
Comment (spericas) [reply webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java:48]: You mean a cache of size less than 1 KB?

File: docs/src/main/asciidoc/mp/server.adoc (hunk 1 above)
Comment (spericas) [reply docs/src/main/asciidoc/mp/server.adoc:354]: No, I meant that "list-based" reads better in the English sentence

**Review by spericas — APPROVED** [review]
All my comments are pretty minor, no need to block this PR

File: webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java (hunk 5 above)
Comment (tomas-langer) [reply webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java:48]: for example in tests... But maybe we could go with kb. I will revisit it when fixing your other comments.

Comment (tomas-langer) [reply webserver/static-content/src/main/java/io/helidon/webserver/staticcontent/MemoryCacheConfigBlueprint.java]: I have done (I hope) the right thing:
- introduced `io.helidon.common.Size`
- updated builders to support it
- updated PR to use it

Size supports:
- `kB` or `kb` for decimal sizes (powers of 1000)
- `KB` or `KiB` for binary sizes (powers of 1024)
- support from kilobytes to exabytes (kilo, mega, giga, tera, peta, exa)

Appropriate parse methods and create methods added.

**Review by spericas — APPROVED** [review]
LGTM

**Pull request merged**